internal/applicationsnapshot/report.go

```diff
@@ -132,3 +172,94 @@ func condensedMsg(results []output.Result) map[string][]string {
 	}
 	return shortNames
 }
+
+// toHacbs returns a version of the report that conforms to the
+// HACBS_TEST_OUTPUT format.
+func (r *Report) toHacbs() hacbsReport {
+	result := hacbsReport{
+		Namespace: application_snapshot_image.ConftestNamespace,
+		Timestamp: r.created,
+	}
+
+	hasFailures := false
+	for _, component := range r.Components {
+		result.Failures += len(component.Violations)
+		result.Warnings += len(component.Warnings)
+		if component.Success {
+			result.Successes += 1
+		} else {
+			// It is possible, although quite unusual, that a component has no
+			// listed violations but is still marked as not successful.
+			hasFailures = true
+		}
+	}
+
+	switch {
+	case result.Failures > 0 || hasFailures:
+		result.Result = "FAILURE"
+	case result.Warnings > 0:
+		result.Result = "WARNING"
+	case result.Successes == 0:
+		result.Result = "SKIPPED"
+	default:
+		result.Result = "SUCCESS"
+	}
+
+	return result
+}
+
+type reportWriter struct {
+	format string
+	writer io.Writer
+}
+
+// Write converts the report to a specific format, and writes it out.
+func (o *reportWriter) Write(report Report) error {
+	data, err := report.toFormat(o.format)
+	if err != nil {
+		return err
+	}
+	_, err = o.writer.Write(data)
+	return err
+}
+
+// newReportWriter creates a new instance of reportWriter from a given target.
+// If a filename is not provided in the target, the defaultWriter is used.
+// If a format is not specified, JSON is used.
+func newReportWriter(target string, defaultWriter io.Writer) (reportWriter, error) {
+	var format, path string
+	parts := strings.SplitN(target, "=", 2)
+	switch len(parts) {
+	case 1:
+		format = parts[0]
+	case 2:
+		format = parts[0]
+		path = parts[1]
+	}
+
+	if len(format) == 0 {
+		format = JSON
+	}
```

Comment:
nitpickery

```suggestion
	parts := strings.SplitN(target, "=", 2)
	var format, path string
	switch len(parts) {
		case 0:
			format = JSON
		case 1:
			format = parts[0]
		case 2:
			format = parts[0]
			path = parts[1]
	}
```

Reply:
That's what I had initially, but then I decided to test this: `--output=`. This hits `case 1` but sets the format to an empty string.